src/test/java/com/googleinterns/zoomtube/servlets/TranscriptLanguageServletTest.java

```diff
@@ -0,0 +1,123 @@
+
+  @Before
+  public void setUp() throws Exception {
+    transcriptLanguageServlet = new TranscriptLanguageServlet();
+    transcriptLanguageServlet.init();
+    transcriptLanguages = new StringWriter();
+    PrintWriter writer = new PrintWriter(transcriptLanguages);
+    when(response.getWriter()).thenReturn(writer);
+  }
+
+  @Test
+  public void doGet_missingVideoLink_badRequest() throws Exception {
+    transcriptLanguageServlet.doGet(request, response);
+
+    verify(response).sendError(
+        HttpServletResponse.SC_BAD_REQUEST, /* message= */ "Missing link parameter.");
+  }
+
+  @Test
+  public void doGet_invalidVideoLink_badRequest() throws Exception {
+    when(request.getParameter(TranscriptLanguageServlet.PARAM_LINK)).thenReturn("123456");
+
+    transcriptLanguageServlet.doGet(request, response);
+
+    verify(response).sendError(
+        HttpServletResponse.SC_BAD_REQUEST, /* message= */ "Invalid video link.");
+  }
+
+  @Test
+  public void doGet_getListOfLanguages_videoWithManyLanguages()
+      throws Exception {
+    String videoWithManyLanguageOptions = "https://www.youtube.com/watch?v=fzQ6gRAEoy0";
+    when(request.getParameter(TranscriptLanguageServlet.PARAM_LINK))
+        .thenReturn(videoWithManyLanguageOptions);
+
+    transcriptLanguageServlet.doGet(request, response);
+
+    List<TranscriptLanguage> actualTranscriptLanguages =
+        transcriptLanguages(transcriptLanguages.toString());
+    assertThat(actualTranscriptLanguages.size()).isEqualTo(38);
+  }
+
+  @Test
+  public void doGet_getListOfLanguages_videoWithNoLanguages() throws Exception {
+    String videoWithNoLanguageOptions = "https://www.youtube.com/watch?v=QJO3ROT-A4E";
+    when(request.getParameter(TranscriptLanguageServlet.PARAM_LINK))
+        .thenReturn(videoWithNoLanguageOptions);
+
+    transcriptLanguageServlet.doGet(request, response);
+
+    List<TranscriptLanguage> actualTranscriptLanguages =
+        transcriptLanguages(transcriptLanguages.toString());
+    assertThat(actualTranscriptLanguages.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void doGet_getListOfLanguages_videoWithOnlyOneLanguage()
+      throws Exception {
+    String videoWithOnlyOneLanguage = "https://www.youtube.com/watch?v=9DwzBICPhdM";
+    when(request.getParameter(TranscriptLanguageServlet.PARAM_LINK))
+        .thenReturn(videoWithOnlyOneLanguage);
+
+    transcriptLanguageServlet.doGet(request, response);
+
+    List<TranscriptLanguage> actualTranscriptLanguages =
+        transcriptLanguages(transcriptLanguages.toString());
+    assertThat(actualTranscriptLanguages.size()).isEqualTo(1);
+  }
+
+  private static List<TranscriptLanguage> transcriptLanguages(String transcriptLanguagesJson) {
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    return (ArrayList<TranscriptLanguage>) gson.fromJson(
+        transcriptLanguagesJson, (new ArrayList<List<TranscriptLanguage>>().getClass()));
```

Comment:
Can you split this return into 2 lines (moving the `Type` to its own line/variable)?

Reply:
^^

Reply:
done!